tap_impact/sync.py

```diff
@@ -280,48 +287,56 @@ def sync_endpoint(client,
-                                    .format(child_stream_name, stream_name, parent_id))
-                            child_path = child_endpoint_config.get(
-                                'path', child_stream_name).format(str(parent_id))
-                            child_bookmark_field = next(iter(child_endpoint_config.get(
-                                'replication_keys', [])), None)
-                            child_total_records = sync_endpoint(
-                                client=client,
-                                catalog=catalog,
-                                state=state,
-                                start_date=start_date,
-                                stream_name=child_stream_name,
-                                path=child_path,
-                                endpoint_config=child_endpoint_config,
-                                static_params=child_endpoint_config.get('params', {}),
-                                bookmark_query_field=child_endpoint_config.get(
-                                    'bookmark_query_field'),
-                                bookmark_field=child_bookmark_field,
-                                bookmark_type=child_endpoint_config.get('bookmark_type'),
-                                data_key=child_endpoint_config.get('data_key', 'results'),
-                                id_fields=child_endpoint_config.get('key_properties'),
-                                selected_streams=selected_streams,
-                                parent=child_endpoint_config.get('parent'),
-                                parent_id=parent_id)
-                            LOGGER.info(
-                                'FINISHED Sync for Stream: {}, parent_id: {}, total_records: {}'\
-                                    .format(child_stream_name, parent_id, child_total_records))
+                        model_id = config.get('model_id')
+                        process_child = True
+                        # conversion_paths endpoint requires model_id tap config param
+                        if child_stream_name == 'conversion_paths' and not(model_id):
+                            process_child = False
+                        if process_child: 
+                            write_schema(catalog, child_stream_name)
+                            # For each parent record
+                            for record in transformed_data:
+                                i = 0
+                                # Set parent_id
+                                for id_field in id_fields:
+                                    if i == 0:
+                                        parent_id_field = id_field
+                                    if id_field == 'id':
+                                        parent_id_field = id_field
+                                    i = i + 1
+                                parent_id = record.get(parent_id_field)
```

Comment:
Correct me if I'm wrong:
No matter what is in `id_fields`, `parent_id_field` is going to be `id_fields[0]` or `"id"`

So couldn't this loop go away and be replaced with `parent_id = record.get(id_fields[0]) or record.get('id')`?

Reply:
The singer spec has us define key_properties as a set, because an endpoint could have multiple key fields.
Most of the endpoints in this tap have a single field key. And most of those (but not all) are 'id'.
Example: 'key_properties': ['id'] - but there could be multiple fields in that set. If there are multiple fields, then the most important one is the 'id' field. It chooses the first field in the ID list, then if 'id' happens to be in that list, it uses that one instead.

This part of the code deals with parent endpoints that have child endpoints, where the child's URL contains the parent's ID field. 99% of the time, this is only a single field, and it is usually called 'id' - but not always.
Anyway, this part of the code works. This code didn't really change, I just had to indent it to add another condition above. It all looks new, but it didn't really change (except for indenting).